service/src/test/kotlin/app/cash/backfila/actions/StartStopBackfillActionTest.kt

```diff
@@ -27,128 +30,162 @@ import misk.hibernate.load
+
+  @BeforeEach
+  fun setup() {
+    // Always configure the backfills before any test
     scope.fakeCaller(service = "deep-fryer") {
       configureServiceAction.configureService(
         ConfigureServiceRequest.Builder()
           .backfills(
             listOf(
-              ConfigureServiceRequest.BackfillData(
-                "ChickenSandwich", "Description", listOf(), null,
-                null, false
-              )
+              ConfigureServiceRequest.BackfillData.Builder()
+                .name("ChickenSandwich")
+                .description("Description")
+                .build(),
+              ConfigureServiceRequest.BackfillData.Builder()
+                .name("BeefSandwich")
+                .description("Description")
+                .build()
             )
           )
           .connector_type(Connectors.ENVOY)
           .build()
       )
     }
+  }
+
+  @Test fun `Backfila starts with no backfills`() {
     scope.fakeCaller(user = "molly") {
       var backfillRuns = getBackfillRunsAction.backfillRuns("deep-fryer")
-      assertThat(backfillRuns.paused_backfills).hasSize(0)
-      assertThat(backfillRuns.running_backfills).hasSize(0)
+      softAssert {
+        assertThat(backfillRuns.paused_backfills).hasSize(0)
+        assertThat(backfillRuns.running_backfills).hasSize(0)
+      }
+    }
+  }
 
-      val response = createBackfillAction.create(
-        "deep-fryer",
-        CreateBackfillRequest.Builder()
-          .backfill_name("ChickenSandwich")
-          .build()
-      )
+  @Nested inner class `After a backfill is created` {
+    var createdBackfillRunId: Long = 0 // lateinit won't work on primitives
 
-      backfillRuns = getBackfillRunsAction.backfillRuns("deep-fryer")
-      assertThat(backfillRuns.paused_backfills).hasSize(1)
-      assertThat(backfillRuns.running_backfills).hasSize(0)
+    @BeforeEach fun createBackfill() {
+      scope.fakeCaller(user = "molly") {
+        val response = createBackfillAction.create(
+          "deep-fryer",
+          CreateBackfillRequest.Builder()
+            .backfill_name("ChickenSandwich")
+            .build()
+        )
+        createdBackfillRunId = response.backfill_run_id
+      }
+    }
 
-      val id = response.backfill_run_id
-      assertThat(backfillRuns.paused_backfills[0].id).isEqualTo(id.toString())
-      startBackfillAction.start(id, StartBackfillRequest())
+    @Test fun `The backfill starts out paused`() {
+      scope.fakeCaller(user = "molly") {
+        var backfillRuns = getBackfillRunsAction.backfillRuns("deep-fryer")
+        softAssert {
+          assertThat(backfillRuns.paused_backfills).hasSize(1)
+          assertThat(backfillRuns.running_backfills).hasSize(0)
+          assertThat(backfillRuns.paused_backfills).singleElement().extracting { it.id }
+            .isEqualTo(createdBackfillRunId.toString())
+        }
+      }
+    }
+
+    @Nested inner class `And then the backfill is started` {
```

Comment:
what is this?

Reply:
Yeah this is where I overdid it a bit I think. I was trying out some new stuff.

Reply:
lol it's a lil awkward to me, can we just call `setupOtherThing()` as the first line in the tests that need extra setup?